packages/playwright-core/src/utils/wsServer.ts

```diff
@@ -77,8 +77,12 @@ export class WSServer {
           reject(new Error('Could not bind server socket'));
           return;
         }
-        const wsEndpoint = typeof address === 'string' ? `${address}${path}` : `ws://${hostname || 'localhost'}:${address.port}${path}`;
-        resolve(wsEndpoint);
+        if (typeof address === 'string') {
+          resolve(`${address}${path}`);
+          return;
+        }
+        const resolvedHost = address.family === 'IPv4' ? address.address : `[${address.address}]`;
```

Comment:
- Shouldn't this be `hostname` when it is defined?
- I am afraid this could accidentally break someone expecting `localhost` in the address and getting `127.0.0.1` or `[::1]` instead. Should we wait for a bug report?